src/main/resources/public/ts/sniplets/workspace-nextcloud-folder.sniplet.ts

```diff
@@ -103,13 +111,16 @@ class ViewModel implements IViewModel {
             isSelectedFolder(folder: models.Element): boolean {
                 return viewModel.selectedFolder === folder;
             },
-            async openFolder(folder: models.Element): Promise<void> {
+            setSelectedFolder(folder : models.Element): void {
+                viewModel.selectedFolder = folder;
+            },
+            async openFolder(folder: any | models.Element): Promise<void> {
                 viewModel.setSwitchDisplayHandler();
-
                 // create handler in case icon are only clicked
-                viewModel.watchFolderState();
                 viewModel.selectedFolder = folder;
+                viewModel.watchFolderState();
                 if (!viewModel.openedFolder.some((openFolder: models.Element) => openFolder === folder)) {
+                    viewModel.openedFolder = viewModel.openedFolder.filter(e => e.path != folder.path);
```

Comment:
tu peux pas le typer à cause de l'erreur "path no existant in SyncDocument" ?

Reply:
Tu peux le caster dans ton callback de ton filter au lieu de dire dans ton interface que ca peut être un "any"